skopt/space.py

```diff
@@ -246,13 +264,23 @@ def __init__(self, categories, prior=None):
         self.categories = categories
         self.transformer = _CategoricalEncoder()
         self.transformer.fit(self.categories)
+        self._prior = prior
 
         if prior is None:
             prior = np.tile(1. / len(self.categories), len(self.categories))
 
         # XXX check that sum(prior) == 1
         self._rvs = rv_discrete(values=(range(len(self.categories)), prior))
 
+    def __eq__(self, other):
+        return (type(self) is type(other)
+                and self.categories == other.categories
+                and self._prior == other._prior)
+
+    def __repr__(self):
+        return "Categorical(categories={}, prior={})".format(
+            self.categories, self._prior)
```

Comment:
I have a minor concern here. If there are too many categories, this would not toll great right?

Should we just print the first say 3 - 4 and the last 3 - 4 categories?